### PR TITLE
16 avatar

### PR DESCRIPTION
**Pull request by tsiguenz**

### CODE REVIEW

File: backend/src/users/users.service.ts [hunk 1]
```diff
@@ -51,24 +35,23 @@ export class UsersService {
...
+        avatarPath: true,
         createdAt: true
       }
     });
     return users;
   }
 
   //TODO: Use guard to verify authorization
-  async deleteUser(paramName: string, nickname: string) {
+  async deleteUser(paramName: string, nickname: string, avatarPath: string) {
```
Comment (Crysicia) [inline backend/src/users/users.service.ts:48]: Can't we call deleteUser only with userId or nickname?
Seems tedious to retrieve every resources to be deleted before calling this function.

File: backend/src/profile/profile.controller.ts [hunk 2]
```diff
@@ -29,12 +42,37 @@ export class ProfileController {
...
+            .map(() => Math.round(Math.random() * 16).toString(16))
+            .join('');
+          return cb(null, `/${randomName}${extname(file.originalname)}`);
+        }
+      })
+    })
+  )
+  @Post('avatar')
+  // TODO: pass the user in the url (/api/users/:nickname/avatar)
+  async uploadAvatar(
```
[the remark this answers is not in the document]
Comment (tsiguenz) [reply backend/src/profile/profile.controller.ts:72]: For the moment only the current user can change his avatar.
When #27 will be done, we will use guard to allow the user to change only his avatar.

File: frontend/src/views/Profile.vue [hunk 3]
```diff
@@ -86,8 +93,13 @@ export default {
         this.user = response.data;
         this.newNickname = this.user.nickname;
         this.newTwoFactorEnable = this.user.twoFactorEnable;
+        this.avatarPath = constants.API_URL + this.user.avatarPath;
+        this.avatarPath = this.avatarPath.replace('/api', '');
```
[the remark this answers is not in the document]
Comment (tsiguenz) [reply frontend/src/views/Profile.vue]: fix here [8e96157](https://github.com/tsiguenz/ft_transcendence/pull/40/commits/8e9615749fed3e6879fbb8665d874f2be8b761c3)

**Review by Crysicia — APPROVED** [review]
🙌

**Pull request merged**